### PR TITLE
Andy/adding auth

### PR DESCRIPTION
**Pull request by andy-little**
## What?

- improved error handling
- added user model to database
- allow user to sign up and sign in
- protected api routes
- decode user info from authorisation header and add to userContext on req

### CODE REVIEW

**Comment by andy-little** [comment]
@TomF-K when you review this you want to click on the `files changed` tab and then click `review changes` and then approve, rather than merging the pull request. Once you've approved I'll merge it when ready :) 

Screenshot 2022-12-31 at 10 35 18

**Review by TomF-K — APPROVED** [review]

**Pull request merged**
